File: src/services/webapi/src/modules/posts/dto/PostCreate.dto.ts
import { Type } from 'class-transformer';
import {
  IsNotEmpty,
  IsNotEmptyObject,
  IsNumber,
  IsString,
  MinLength,
  ValidateNested,
} from 'class-validator';
import { AddressCreateDto } from './AddressCreate.dto';

export class PostCreateDto {
  @IsNotEmpty()
  @IsString()
  @MinLength(3)
  title: string;
  @IsNotEmpty()
  @IsString()
  @MinLength(5)
  content: string;
  @IsNotEmptyObject()
  @ValidateNested()
  @Type(() => AddressCreateDto)
  address: AddressCreateDto;
  userId?: number = null;
}

File: src/clients/android/app/src/main/java/com/example/student_community/models/webapi/dto/PostCreateDto.kt
package com.example.student_community.models.webapi.dto

import com.google.gson.annotations.SerializedName

data class PostCreateDto(
    @SerializedName("title") val Title: String,
    @SerializedName("content") val Content: String,
    @SerializedName("address") var Address: AddressCreateDto,
)